nixos/modules/system/etc/build-composefs-dump.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+
+"""Build a composefs dump from a Json config"""
+
+import functools
+import glob
+import json
+import os
+import sys
+from enum import Enum
+from typing import Any
+
+Attrs = dict[str, Any]
+
+
+class FileType(Enum):
+    directory = "4"
+    file = "10"
+    symlink = "12"
+
+
+class ComposefsPath:
```

Comment:
Maybe a good candidate for `@dataclass` but feel free to ignore if not appropriate.

Reply:
It used to be a dataclass but since I provide my own constructor and don't use __repr__, the dataclass just doesnt provide anything useful.